Yoast/wordpress-seo
# Create Structured Data Block - FAQ Page

## Summary

This PR can be summarized in the following changelog entry:

* Added a Yoast Structured Data block for creating FAQ pages.

## Relevant technical choices:

* The code has swollen slightly for this implementation, and could definitely use a cleaning-pass, but it works and it needs to be in 8.1

## Test instructions

This PR can be tested by following these steps:

* Make sure you have the feature flag set to true:
```define('YOAST_FEATURE_GUTENBERG_STRUCTURED_DATA_BLOCKS', true );```
* Add the FAQ block in a post in the Gutenberg editor.
* Verify that everything functions as you expect: Editing title and questions/answers, inserting, removing, moving, etc.
* Verify that editting triggers autosaving correctly (there are no "Update failed" warnings).
* Use the Ordered/Unordered list toggle in the sidebar, and verify that the block switches out numbers for bullet points and vice-versa as expected.
* Save the post and view it in the front-end.
* Verify that the HTML structure is as you'd expect (h2, h3, p tags). Also pay special mind to `<ul>` or `<ol>` being there depending on whether you selected unorderedlist or not, respectively.
* Inspect the HTML in console, locate this json_ld script tag above the FAQ block's title:
![image](https://user-images.githubusercontent.com/26220788/44155284-2d31154a-a0ad-11e8-8d97-99ab08f8a44e.png)

* Copy-paste what you see inside that tag:
![image](https://user-images.githubusercontent.com/26220788/44155340-48ca1f0e-a0ad-11e8-8ec6-e9d773654666.png)

And paste to https://search.google.com/structured-data/testing-tool/u/0/ as a code snippet. Verify that there are no errors and everything is correctly identified as a FAQPage and Question with AcceptedAnswer.


## Quality assurance


Fixes https://github.com/Yoast/yoast.com/issues/3145
Fixes https://github.com/Yoast/wordpress-seo/issues/10560
